### DISCUSSION TITLE
how to keep the same structure when copy the old mdf to a new mdf

### DISCUSSION THREAD

**Discussion by hljjj** [danielhrisca/asammdf#786] [Q&A]
hello,

I need to edit the data in a datalist from a md4 file and repack the result, I using mdf.append to copy old channel and add new_channel, But in the end the new mdf file has a different structure from the old mdf file, how to keep the same structure?

the old structurel look like this:
![original](https://user-images.githubusercontent.com/45890184/201622938-8371980f-b0a5-4387-9904-926bea9cd275.png)

the new structurel  become this:
<img width="779" alt="new" src="https://user-images.githubusercontent.com/45890184/201623059-9eb24799-1b52-49bf-9b38-3838c1eb1ba7.png">

here is my code
```python
origin_mdf_path = 'c:/files/data/origin.mf4'
new_mdf_path = 'c:/files/data/new_mdf.mf4'

mdf = MDF(origin_mdf_path)
new_mdf = MDF()

for m in mdf.iter_channels():
    if m.name != 'VideoRawdata_VC0':
        new_mdf.append(m)
    else:
        new_samples = rebuild_samples(m.samples)
        new_signal = Signal(
            samples=new_samples,
            timestamps=m.timestamps,
            unit=m.unit,
            name=m.name,
            conversion=m.conversion,
            raw=m.raw,
            master_metadata=m.master_metadata,
            display_names=m.display_names,
            attachment=m.attachment,
            stream_sync=m.stream_sync,
            invalidation_bits=m.invalidation_bits,
            source=m.source,
            encoding=m.encoding,
            group_index=m.group_index,
            channel_index=m.channel_index)
        new_mdf.append(new_signal)

new_mdf.save(new_mdf_path)
```

Thanks

**Comment by danielhrisca** [comment]
`asammdf` does not use VLSD storage for the channel samples (Signal data blocks list like you see in the first picture). Instead the less complex storage option is used, so the samples are stored directly in the data group records; this is wasting some disk space, but it maps better to the way numpy deals with data.

The end result is the same as far as the user is concerned

**Reply by hljjj** [reply]
Thanks for your reply.

Does it means that I can't keep the same structure if I use asammdf?

By the way, are there any way to reset the index of the channel?

In the new mdf file,  the index of 'VideoRawdata_VC0' is (1,1)，I want to reset to (0,2)

**Reply by hljjj** [reply]
I find the way to repack channels to the same data group, like this:

channels = []
for m in mdf.iter_channels():
    if m.name != 'VideoRawdata_VC0':
       channels.append(m)
    else:
        new_samples = rebuild_samples(m.samples)
        new_signal = Signal(
            samples=new_samples,
            timestamps=m.timestamps,
            unit=m.unit,
            name=m.name,
            conversion=m.conversion,
            raw=m.raw,
            master_metadata=m.master_metadata,
            display_names=m.display_names,
            attachment=m.attachment,
            stream_sync=m.stream_sync,
            invalidation_bits=m.invalidation_bits,
            source=m.source,
            encoding=m.encoding,
            group_index=m.group_index,
            channel_index=m.channel_index)
        channels.append(new_signal)

new_mdf.append(signals)
